src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.LocalFunctions.cs

```diff
@@ -124,17 +124,29 @@ public override BoundNode VisitLocalFunctionStatement(BoundLocalFunctionStatemen
 
             var oldPending = SavePending(); // we do not support branches into a lambda
 
-            // Local functions don't affect outer state and are analyzed
-            // with everything unassigned and reachable
+            // SPEC: The entry point to a local function is always reachable.
+            // Captured variables are assigned if they are assigned on all
+            // branches into the local function.
+
             var savedState = this.State;
             this.State = this.ReachableState();
 
-            var usages = GetOrCreateLocalFuncUsages(localFuncSymbol);
-            var oldReads = usages.ReadVars;
-            usages.ReadVars = BitVector.Empty;
-
             if (!localFunc.WasCompilerGenerated) EnterParameters(localFuncSymbol.Parameters);
 
+            // Captured variables are definitely assigned if they are assigned on
+            // all branches into the local function, so we store all reads from
+            // possibly unassigned captured variables and later report definite
+            // assignment errors if any of the captured variables is not assigned
+            // on a particular branch.
+            // Assignments to captured variables are also recorded, as a local function
+            // definitely assigns captured variables on a call to a local function
+            // if that variable is definitely assigned at all branches out of the
```

Comment:
>branches out of [](start = 62, length = 15)

"branches out of" should be "returns from".

Reply:
Yield return, await, and the implicit return before the first statement in an iterator all count -- would you classify those as return statements?